firmware/src/widgets/stockWidget.cpp

```diff
@@ -108,29 +112,66 @@ void StockWidget::displayStock(int8_t displayIndex, StockDataModel &stock, uint3
     TFT_eSPI &display = m_manager.getDisplay();
 
     display.fillScreen(TFT_BLACK);
-    display.setTextColor(TFT_WHITE);
-    display.setTextSize(4);  // Set text size
+
 
     // Calculate center positions
     int screenWidth = display.width();
     int centre = 120;
-
-    // Draw stock data
-    display.fillRect(0, 0, screenWidth, 50, 0x0256);  // rgb565 colors
-    display.drawString(stock.getSymbol(), centre, 27, 1);
-    display.drawString("$" + stock.getCurrentPrice(2), centre, 51 + display.fontHeight(1), 1);
-
+    int arrowOffsetX = 0;
+    int arrowOffsetY = -109;
+
+    // ensuring proper currency display
+    String currencySymbol = "$";
+    String currencySymbolEUR = "EUR";
+    String currencySymbolGBP = "GBP";
+    // the below are just placeholders untill we have a unicode font that supports the currency symbols, which is why we're not just using the direct value from getCurrency.
+    // additionally, in some cases(crypto) the ticker wont return a currency value, so it must be extracted from the symbol deffinition 
+    if (stock.getCurrency() == "EUR"){
+       currencySymbol = currencySymbolEUR;
+    } else if (stock.getCurrency() == "GBP"){
+       currencySymbol = currencySymbolGBP;
+    } else if (stock.getSymbol().indexOf("/EUR") != -1){
+       currencySymbol = currencySymbolEUR;
+    } else if (stock.getSymbol().indexOf("/GBP") != -1){
+       currencySymbol = currencySymbolGBP;
+    }
+// Outputs
+    display.fillRect(0,70, screenWidth, 49, TFT_WHITE); 
+    display.fillRect(0,111, screenWidth, 15, TFT_LIGHTGREY);  
+    display.setTextSize(1);  
+    display.setTextColor(TFT_WHITE);
+    display.drawString("52 Week:", centre, 188, 1);
```

Comment:
Nit: Shouldn't this be `52 Weeks:` ?
(Note that I am not a native English speaker, so I might be wrong here)

Reply:
Valid Nit, and you are correct technically.
However in the world of stocks the colloquialism/widely accepted term is as written (also applies to 3 month low/high etc)

Classic english and its weird edge cases

Reply:
I believe the reason is that the full version would be something like "52 week high". Which is the highest value from the last 52 weeks. So you are still only referencing one value therefore it is singular.